src/main/java/com/googlecode/protobuf/format/JsonFormat.java

```diff
@@ -1510,4 +1590,48 @@ private static String prepareNumberFromString(String numberText) {
 
         return numberText;
     }
+    
+    /**
+     * @return true if input String can be parsed as 32-bit hex
+     */
+    private static boolean is32BitHex(String value) {
+    	return value.matches("0x[0-9a-fA-F]{8}");    	
+    }
+    
+    /**
+     * @return true if input String can be parsed as 64-bit hex
+     */
+    private static boolean is64BitHex(String value) {
+    	return value.matches("0x[0-9a-fA-F]{16}");    	
+    }
+     
+    private static float parseFloatOrFixed32(String value) {    	
+    	try {
+    		return(parseFloat(value));
+    	} catch (NumberFormatException e) {
+    		int intValue = parseInt32(value);    		
+    		ByteBuffer byteBuffer = (ByteBuffer)ByteBuffer
+        			.allocate(Integer.BYTES)
+        			.putInt(intValue)
+        			.order(ByteOrder.BIG_ENDIAN)
+        			.rewind();
+    		float f = byteBuffer.getFloat();
+    		return f;
+    	}
+    }
+    
+    private static double parseDoubleOrFixed64(String value) {
+    	try {
+    		return(parseDouble(value));
+    	} catch (NumberFormatException e) {
+    		long longValue = parseInt64(value);    		
+    		ByteBuffer byteBuffer = (ByteBuffer)ByteBuffer

```

Comment:
If you need to, you can also use Long.reverseBytes(longValue)

Something like `return Double.longBitsToDouble(Long.longBitsToDouble(longValue))`